lib/resty/core/phase.lua

```diff
@@ -0,0 +1,55 @@
+local ffi = require 'ffi'
+local C = ffi.C
+local base = require "resty.core.base"
+local FFI_ERROR = base.FFI_ERROR
+local tostring = tostring
+
+ffi.cdef[[
+int ngx_http_lua_ffi_get_phase(ngx_http_request_t *r, char **err)
+]]
+
+
+local _M = {
+    _VERSION = base.version
+}
+
+local errmsg = base.get_errmsg_ptr()
+local context_lookup = {}
+context_lookup[0x0001] = "set"
+context_lookup[0x0002] = "rewrite"
+context_lookup[0x0004] = "access"
+context_lookup[0x0008] = "content"
+context_lookup[0x0010] = "log"
+context_lookup[0x0020] = "header_filter"
+context_lookup[0x0040] = "body_filter"
+context_lookup[0x0080] = "timer"
+context_lookup[0x0100] = "init_worker"
+context_lookup[0x0200] = "balancer"
+context_lookup[0x0400] = "ssl_cert"
+context_lookup[0x0800] = "ssl_session_store"
+context_lookup[0x1000] = "ssl_session_fetch"
```

Comment:
I am curious as to why this form is being used to assign values to this table? Isn't this triggering 4 rehashes of the table? Additionally, isn't it messier than simply:
```lua
local context_lookup = {
    [0x0001] = "set",
    ...
}
```

And last of all, I am unsure about this but shouldn't the name be capitalized, as this is meant to be a constant? I am not familiar with the preferred naming style with such "constant tables".